lib/git/diff.rb

```diff
@@ -89,10 +89,10 @@ def binary?
       end
 
       def blob(type = :dst)
-        if type == :src
-          @base.object(@src) if @src != '0000000'
-        else
-          @base.object(@dst) if @dst != '0000000'
+        if type == :src && !/0{7,40}/.match(@src)
+          @base.object(@src)
+        elsif !/0{7,40}/.match(@dst)
```

Comment:
Might be good to pull the regex out to a constant and use `match?` (instead of `match`) since you don't care to capture any values.

Reply:
Updated to use a constant for the regex. I initially tried `match?` but that isn't available in Ruby 2.3 which is in the test matrix for this project.